pkg/vpcmodel/mdOutput.go

```diff
@@ -11,35 +11,60 @@ type MDoutputFormatter struct {
 }
 
 const (
-	mdTitle  = "## Endpoint connectivity report"
-	mdHeader = "| src | dst | conn |\n|-----|-----|------|"
+	mdDefaultTitle        = "## Endpoint connectivity report"
+	mdDefaultHeader       = "| src | dst | conn |\n|-----|-----|------|"
+	mdEndpointsDiffTitle  = "## Endpoints diff report"
+	mdSubnetsDiffTitle    = "## Subnets diff report"
+	mdEndPointsDiffHeader = "| type | src |  dst | conn1 | conn2 | vsis-diff-info |\n" +
+		"|------|-----|------|-------|-------|----------------|"
+	mdSubnetsDiffHeader = "| type | src |  dst | conn1 | conn2 | subnets-diff-info |\n" +
+		"|------|-----|------|-------|-------|-------------------|"
 )
 
 func (m *MDoutputFormatter) WriteOutput(c1, c2 *VPCConfig,
 	conn *VPCConnectivity,
 	subnetsConn *VPCsubnetConnectivity,
-	subnetsDiff *diffBetweenCfgs,
+	cfgsDiff *diffBetweenCfgs,
 	outFile string,
 	grouping bool,
-	uc OutputUseCase) (*VPCAnalysisOutput, error) {
+	uc OutputUseCase) (*SingleAnalysisOutput, error) {
 	// get output by analysis type
-	out := "# " + headerOfAnalyzedVPC(c1.VPC.Name(), "")
+	v2Name := ""
+	if c2 != nil {
+		v2Name = c2.VPC.Name()
+	}
+	out := "# " + headerOfAnalyzedVPC(uc, c1.VPC.Name(), v2Name)
 	switch uc {
 	case AllEndpoints:
-		lines := []string{mdTitle, mdHeader}
+		lines := []string{mdDefaultTitle, mdDefaultHeader}
 		connLines := m.getGroupedOutput(conn)
 		sort.Strings(connLines)
 		lines = append(lines, connLines...)
 		out += strings.Join(lines, "\n")
 		out += asteriskDetails
+	case SubnetsDiff, EndpointsDiff:
+		var mdTitle, mdHeader string
+		if uc == EndpointsDiff {
+			mdTitle = mdEndpointsDiffTitle
+			mdHeader = mdEndPointsDiffHeader
+		} else {
+			mdTitle = mdSubnetsDiffTitle
+			mdHeader = mdSubnetsDiffHeader
+		}
+		lines := []string{mdTitle, mdHeader}
+		connLines := m.getGroupedDiffOutput(cfgsDiff)
+		sort.Strings(connLines)
+		lines = append(lines, connLines...)
+		out += strings.Join(lines, "\n")
+		out += asteriskDetails
```

Comment:
some common code with `AllEndpoints` can be moved to a function, to avoid duplication.

Reply:
done